### DISCUSSION TITLE
Documentation about interacting with a Gleam program

### DISCUSSION THREAD

**Discussion by tkivela** [gleam-lang/gleam#2868] [Ideas & suggestions]
I'm learning the language and I think Gleam Language Tour is just great. Normally when I'm learning a new language I like to start with "Hello world" kind of program (which is beatifully created with your documentation by running "gleam new && gleam run"). However from that point onwards I think the documentation could give some pointers how to make the program interactive, mainly how to get input from the user. I first thought that the standard library's io -package would have some stdio reading functions but it didn't (previous languages that I've worked with had that).

I know that there's some Erlang modules interacting with stdio and for example reading files and I think it would benefit new developers greatly if the official site gave some pointers where to look things (even if they're not part of the actual language core or officially maintained by the Gleam team).

Some pointers for these things would be great:
- Reading from standard input (getting user's input from the terminal)
- Reading and writing from files
- Calling http endpoints (async calls / tasks)

These would get people a long way! If there is already documentation and examples of these on the site, I'm sorry for not finding them. If there isn't, I think it would be a great addition to the site.

Thank you!

**Comment by lpil** [comment]
Hello there! Yes, I do agree. The plan is to have a "cookbook" or such, with lots of guides on how to perform various tasks in Gleam.
